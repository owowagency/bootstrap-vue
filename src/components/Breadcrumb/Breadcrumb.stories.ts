import Breadcrumb from '.';
import BreadcrumbItem from '@/components/BreadcrumbItem';

export default {
    title: 'Components/Breadcrumb',
    component: Breadcrumb,
};

const template = (args) => ({
    components: {Breadcrumb, BreadcrumbItem},
    setup: () => ({args}),
    template: `
        <Breadcrumb v-bind="args">
            <BreadcrumbItem>
                <a href="#">
                    Home
                </a>
            </BreadcrumbItem>

            <BreadcrumbItem>
                <a href="#">
                    Library
                </a>
            </BreadcrumbItem>

            <BreadcrumbItem active>
                Data
            </BreadcrumbItem>
        </Breadcrumb>
    `,
});

export const Default = template.bind({});
