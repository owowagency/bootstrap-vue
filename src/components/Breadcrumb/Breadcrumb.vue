<template>
    <nav>
        <ol class="breadcrumb">
            <!-- @slot Displays content inside the breadcrumb -->
            <slot />
        </ol>
    </nav>
</template>

<docs>
```vue
<Breadcrumb>
    <BreadcrumbItem>
        Home
    </BreadcrumbItem>

    <BreadcrumbItem active>
        Products
    </BreadcrumbItem>
</Breadcrumb>
```
</docs>
