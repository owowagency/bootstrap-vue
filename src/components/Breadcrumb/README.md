# Breadcrumb

```vue
<Breadcrumb>
    <BreadcrumbItem>
        Home
    </BreadcrumbItem>

    <BreadcrumbItem active>
        Products
    </BreadcrumbItem>
</Breadcrumb>
```

## Slots

| Name    | Description                            | Bindings |
| ------- | -------------------------------------- | -------- |
| default | Displays content inside the breadcrumb |          |
