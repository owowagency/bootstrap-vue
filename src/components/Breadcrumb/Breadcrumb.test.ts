import Breadcrumb from '.';
import {describe} from 'vitest';

describe('template', () => {
    componentRenderTest(Breadcrumb);

    componentSlotRenderTest(Breadcrumb);
});
