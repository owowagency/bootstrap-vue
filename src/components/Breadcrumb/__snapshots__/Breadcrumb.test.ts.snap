// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`template > renders default 1`] = `
"<nav>
  <ol class=\\"breadcrumb\\">
    <!-- @slot Displays content inside the breadcrumb -->
  </ol>
</nav>"
`;
